### ISSUE TITLE
can i get 200 results back?

### ISSUE THREAD

**Issue by ralyodio** [TUVIMEN/torge#4]
Some sites support adding a page size option. I know 1337x can give you 200 back at a time.

**Comment by TUVIMEN** [comment]
As of now only `libgen` has `--results` option.

I went to 1337x to find this option but found nothing, could you give me a link to page with 200 results?

**Comment by ralyodio** [comment]
not sure, i was using this module in node and can get 200 back:

https://github.com/Habibouldmoussa/torrent-search-api-for-barbaroussa/tree/f1feeab511bf0821930a43121b8426bbab142fa4

**Comment by TUVIMEN** [comment]
https://github.com/Habibouldmoussa/torrent-search-api-for-barbaroussa/blob/f1feeab511bf0821930a43121b8426bbab142fa4/lib/providers/1337x.js defines `resultsPerPageCount: 20` on line 24

`search` method in https://github.com/Habibouldmoussa/torrent-search-api-for-barbaroussa/blob/f1feeab511bf0821930a43121b8426bbab142fa4/lib/TorrentProvider.js  calls `computePageCount` method to determine how many pages to get so that count of results is at least the `limit` passed to `search`.

If you want 200 results you could run torge in a loop specifying page.

**Comment by ralyodio** [comment]
oh ok, so they are looping.

**Comment by TUVIMEN** [comment]
By all sources do you mean searching query on all torrent sites supported?

**Comment by ralyodio** [comment]
it does, sorry. i meant can we do the searches by sorting by those fields?

**Comment by TUVIMEN** [comment]
Yes you can, when you run `torge -h` you get general help and universal options.

When you run `torge SOURCE -h` you get options specific to your source.

All torrent sources have `se` and `le` sorting e.g. `torge 1337x --json -s se your query`

**Comment by ralyodio** [comment]
ok cool. does it support a --proxy option?

**Comment by TUVIMEN** [comment]
It kind of does, on line 31 there is `ucurl` alias that is universal for every request. You could add this option to it.

**Comment by ralyodio** [comment]
cool

**Comment by TUVIMEN** [comment]
Script hangs because of connection issues, it can be caused by your proxy

**Comment by TUVIMEN** [comment]
`--all` option does not make sense since every source has different options

**Comment by ralyodio** [comment]
ok, your loop i got working with --json and use jq to combine results, however the magnet url is not included in results.

**Comment by ralyodio** [comment]
```
#!/bin/sh

result="[]"

for i in tpb lt nyaa 1337x rarbg
do
    # Append each JSON output directly into the array
    result=$(torge "$i" --no-prompt --json "$1" | jq --argjson arr "$result" '$arr + [.[]]')
done

echo $result
```

**Comment by ralyodio** [comment]
how can i fetch all the magnet urls in one pass?

**Comment by TUVIMEN** [comment]
3 of the sources you use do not give magnet links in search results so no, use `--link-conv` option.

Also getting magnet links specifically is blocked by cloudflare for `rarbg`.

**Comment by ralyodio** [comment]
there's an npm module which gets around cloudflare blocks...the module i sent you earlier uses it -- maybe we can figure out how its working under the hood.

so --link-conv will return magnet links? I think we would need to fetch all pages to get magnet urls. Can we do that?

**Comment by TUVIMEN** [comment]
Yes `--link-conv` fetches all pages to get magnet links, but only when necessery.

**Comment by ralyodio** [comment]
getting this error:

```
$ ./all.sh 'discography'       
curl: option -FLAC-torrent-18549932.html: is badly used here
curl: try 'curl --help' for more information
curl: (2) no URL specified
curl: try 'curl --help' for more information
curl: (2) no URL specified
curl: try 'curl --help' for more information
curl: (2) no URL specified
curl: try 'curl --help' for more information
curl: (2) no URL specified
curl: try 'curl --help' for more information
curl: (2) no URL specified
curl: try 'curl --help' for more information
./all.sh: line 8: /usr/bin/jq: Argument list too long
jq: invalid JSON text passed to --argjson
Use jq --help for help with command-line options,
or see the jq manpage, or online docs  at https://jqlang.github.io/jq

```

which this script:

```
#!/bin/sh

result="[]"

for i in tpb lt nyaa 1337x rarbg
do
    # Append each JSON output directly into the array
    result=$(torge "$i" --no-prompt --link-conv --json "$1" | jq --argjson arr "$result" '$arr + [.[]]')
done

echo $result


```

**Comment by ralyodio** [comment]
i run it with `./all.sh 'discography'`

**Comment by TUVIMEN** [comment]
That npm module is written in javascript (obviously) results send by cloudflare return computations to be made in javascript to bypass it (when it returns capcha you'll have a problem).

**Comment by TUVIMEN** [comment]
There is a slight issue in `limetorrents` which doesn't get all links for results, some are empty and curl returns errors.

**Comment by ralyodio** [comment]
i don't get any results now though.

**Comment by TUVIMEN** [comment]
Because your jq fails

You should have written

```shell
#!/bin/sh

for i in tpb lt nyaa 1337x rarbg
do
    # Append each JSON output directly into the array
    torge "$i" --no-prompt --link-conv --json "$@" 
done | jq '[ .results[] ]'

echo $result
`

**Comment by TUVIMEN** [comment]
`limetorrents` has been [fixed](https://github.com/TUVIMEN/torge/commit/4dff1a1e538a9bb9c11089126ff76c4f313634e6)

**Comment by TUVIMEN** [comment]
This script
```shell
#!/bin/sh

for i in tpb lt nyaa 1337x rarbg
do
    torge "$i" --no-prompt --link-conv --json "$@"
done | jq -s '[ .[].results[] ]'
```
seems to work

**Comment by TUVIMEN** [comment]
although rarbg returns empty links

**Comment by ralyodio** [comment]
oh nice. thanks!!!
